File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/Inbound/BinaryParsing.java
package com.github.grzesiek_galezowski.BastardInjection.Inbound;

import com.github.grzesiek_galezowski.BastardInjection.InMessages.NullMessage;
import com.github.grzesiek_galezowski.BastardInjection.InMessages.StartMessage;
import com.github.grzesiek_galezowski.BastardInjection.InMessages.StopMessage;
import com.github.grzesiek_galezowski.BastardInjection.Interfaces.Message;

class BinaryParsing implements PacketParsing {
  public Message resultFor(byte[] frameData) {
    if (frameData == null) {
      return new NullMessage();
    } else if (frameData[0] == 1) {
      return new StartMessage();
    } else {
      return new StopMessage();
    }

  }
}

File: CSharp/DependencyInjection/DependencyInjection/BastardInjection/_4_BastardInjection/Inbound/BinaryParsing.cs
using BastardInjection._4_BastardInjection.InMessages;
using BastardInjection._4_BastardInjection.Interfaces;

namespace BastardInjection._4_BastardInjection.Inbound
{
  internal interface IPacketParsing
  {
    AcmeMessage ResultFor(byte[] frameData);
  }

  class BinaryParsing : IPacketParsing
  {
    public AcmeMessage ResultFor(byte[] frameData)
    {
      if (frameData == null)
      {
        return new NullMessage();
      }
      else if (frameData[0] == 1)
      {
        return new StartMessage();
      }
      else
      {
        return new StopMessage();
      }
      
    }
  }
}